### PR TITLE
Adding plotly, updating MAGE vignette

### PR DESCRIPTION
**Pull request by Nathaniel-Fernandes**
* Added plotly to `plot_glu` and lasagna plots
* Updated MAGE vignette
* In MAGE, linearly interpolate CGM data to 5 minute intervals using `CGMS2DayByDay`

### CODE REVIEW

File: vignettes/MAGE.Rmd [hunk 1]
```diff
@@ -131,30 +140,206 @@ CGM traces following the original description of MAGE calculations by
...
+also specify the version and release date if we benchmark the proposed
+MAGE algorithm against it, or the reason we were unable to.
+
++-------------+-------------+-------------+-------------+-------------+
+| Tool or     | **I         | Be          | **Version   | Release /   |
+| Algorithm   | mplements   | nchmarked   | Used**      | Update Date |
+| Name        | MAGE**      | Against?    |             |             |
++:===========:+:===========:+:===========:+:===========:+:===========:+
+| iglu        | Yes         | Yes         | iglu v4.1   | Present     |
+| ::mage_ma ( |             |             |             |             |
```
Comment (irinagain) [inline vignettes/MAGE.Rmd]: this has strange new line start

File: vignettes/MAGE.Rmd [hunk 2]
```diff
@@ -131,30 +140,206 @@ CGM traces following the original description of MAGE calculations by
...
+| Proposed)   |             |             |             |             |
++-------------+-------------+-------------+-------------+-------------+
+| iglu::m     | Yes         | Yes         | iglu v2.0   | Feb 27,     |
+| age_naive   |             |             |             | 2020        |
+| (De         |             |             |             |             |
```
Comment (irinagain) [inline vignettes/MAGE.Rmd]: same, change of line

File: vignettes/MAGE.Rmd [hunk 3]
```diff
@@ -131,30 +140,206 @@ CGM traces following the original description of MAGE calculations by
...
+|             |             |             | link for    |             |
+|             |             |             | Version 10  |             |
+|             |             |             | is broken & |             |
+|             |             |             | only        |             |
+|             |             |             | downloads   |             |
+|             |             |             | Version     |             |
+|             |             |             | 9.0)        |             |
++-------------+-------------+-------------+-------------+-------------+
+| cg          | Yes         | Yes         | cg          | Nov 20,     |
+| manalysis   |             |             | manalysis   | 2023        |
```
Comment (irinagain) [inline vignettes/MAGE.Rmd]: change of line

File: vignettes/MAGE.Rmd [hunk 4]
```diff
@@ -131,30 +140,206 @@ CGM traces following the original description of MAGE calculations by
...
++-------------+-------------+-------------+-------------+-------------+
+| CGDA [20]   | Yes         | No:         | N/A         | N/A         |
+|             |             | Randomly    |             |             |
+|             |             | skips       |             |             |
+|             |             | computing   |             |             |
+|             |             | metrics for |             |             |
+|             |             | certain CGM |             |             |
+|             |             | traces      |             |             |
+|             |             | without fl  |             |             |
+|             |             | a g/warning |             |             |
```
Comment (irinagain) [inline vignettes/MAGE.Rmd]: space is already here

File: vignettes/MAGE.Rmd [hunk 5]
```diff
@@ -131,30 +140,206 @@ CGM traces following the original description of MAGE calculations by
...
++-------------+-------------+-------------+-------------+-------------+
+| "Spline     | Yes         | No: noo p   | N/A         | N/A         |
+| Inter       |             | en-source   |             |             |
+| polation"   |             | impl e      |             |             |
+| Method [23] |             | mentation   |             |             |
+|             |             | provided.   |             |             |
+|             |             | Coding from |             |             |
+|             |             | scratch as  |             |             |
+|             |             | per paper   |             |             |
+|             |             | speci f     |             |             |
```
Comment (irinagain) [inline vignettes/MAGE.Rmd]: space is already here

**Review by irinagain — CHANGES_REQUESTED** [review]
see how I indicated spacings in the raw .Rmd

File: vignettes/MAGE.Rmd (hunk 1 above)
Comment (Nathaniel-Fernandes) [reply vignettes/MAGE.Rmd]: I noticed these in the raw text, but when I attempt to make them one-line the table breaks

Comment (Nathaniel-Fernandes) [reply vignettes/MAGE.Rmd]: Perhaps the version of knitr I'm using is incompatible

**Review by Nathaniel-Fernandes — COMMENTED** [review]
I removed the MAGE.Rmd vignette because of formatting issues

**Pull request merged**